File: Find the Difference of Two Arrays/find-the-difference-of-two-arrays.py
class Solution(object):
    def findDifference(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[List[int]]
        """
        set1 = set(nums1)
        set2 = set(nums2)
        
        res1 = [num for num in set1 if num not in set2]
        res2 = [num for num in set2 if num not in set1]
        
        return [res1, res2]

File: Find the Difference of Two Arrays/find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();
        
        for (int num : nums1) {
            set1.add(num);
        }
        for (int num : nums2) {
            set2.add(num);
        }
        
        List<List<Integer>> result = new ArrayList<>();
        List<Integer> ans1 = new ArrayList<>();
        
        for (int num : set1) {
            if (!set2.contains(num)) {
                ans1.add(num);
            }
        }
        result.add(ans1);
        
        List<Integer> ans2 = new ArrayList<>();
        for (int num : set2) {
            if (!set1.contains(num)) {
                ans2.add(num);
            }
        }
        result.add(ans2);
        
        return result;
    }
}
